### PR TITLE
add cleos option to unpack data when forming transaction, useful for …

### PR DESCRIPTION
**Pull request by victorj8**
…example when signing the transaction with trezor wallets

## Change Description

Add an option for cleos to unpack data when forming transaction, useful for example when signing the transaction with trezor wallets #8205 

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: programs/cleos/main.cpp [hunk 1]
```diff
@@ -399,7 +423,13 @@ fc::variant push_transaction( signed_transaction& trx, const std::vector<public_
       }
    } else {
       if (!tx_return_packed) {
-        return fc::variant(trx);
+         if (tx_unpack_data) {
+            fc::variant unpacked_data_trx;
+            abi_serializer::to_variant(trx, unpacked_data_trx, abi_serializer_resolver, abi_serializer_max_time);
```
Comment (heifner) [inline programs/cleos/main.cpp]: Should add a `try-catch` around this and revert to `fc::variant(trx);`. Then I think you can just make that the behavior and remove the new `-u` option.

Comment (victorj8) [reply programs/cleos/main.cpp]: Done

File: tests/nodeos_run_test.py [hunk 2]
```diff
@@ -599,6 +599,33 @@
...
+
+    try:
+        assert(not trans[1]["signatures"])
+    except (AssertionError, KeyError) as _:
+        Print("ERROR: Expected signatures array to be empty due to skipping option enabled.")
+        raise
+
+    try:
+        assert("hex_data" in trans[1]["actions"][0])
+    except (AssertionError, KeyError) as _:
```
Comment (heifner) [inline tests/nodeos_run_test.py]: Can you add checks for `data.from, data.to, data.memo`. We would like to move away from `hex_data` only being there when `data` contains expanded form, so would prefer this test to test for expanded `data` directly.

Comment (victorj8) [reply tests/nodeos_run_test.py]: Done

**Comment by heifner** [comment]
Need to resolve conflict

**Pull request merged**